### PR TITLE
add gitlab-ci pipeline

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
see https://gitlab.com/doodlesbykumbi/conjur-api-go/pipelines

### CODE REVIEW

File: _setup.sh [hunk 1]
```diff
@@ -7,55 +7,80 @@
...
+else
+    CONJUR_DATA_KEY="$(docker-compose run -T --no-deps conjur data-key generate)" \
   docker-compose up --no-deps -d postgres conjur cuke-master
+fi
+
 exec_on conjur conjurctl wait
-exec_on cuke-master /opt/conjur/evoke/bin/wait_for_conjur
+if oss_only; then
+  true
+else
```
Comment (izgeri) [inline _setup.sh]: why not just `if !oss_only` (or whatever the right bash syntax is here)?

File: conjurapi/authn/auth_token_test.go [hunk 2]
```diff
@@ -88,41 +85,6 @@ func TestTokenV4_Parse(t *testing.T) {
...
-		log.Println("Cannot continue.")
-		return
-	}
-
-	if res.StatusCode != http.StatusOK {
-		log.Printf("Received %d response when authenticating.\n", res.StatusCode)
-		log.Println("Cannot continue.")
-	}
-
-	new_token_bytes, _ := ioutil.ReadAll(res.Body)
```
Comment (izgeri) [inline conjurapi/authn/auth_token_test.go]: why is this whole section removed? I'm not sure what is lost by mocking the new token, vs actually retrieving one via this sequence of http requests

File: _setup.sh [hunk 3]
```diff
@@ -7,55 +7,80 @@
...
-for ((i=0; i<$count; i++)); do
-  id="${vars[$i]}"
-  val="${secrets[$i]}"
-  exec_on cuke-master conjur variable create "$id"
-  exec_on cuke-master conjur variable values add "$id" "$val"
-done
-
-api_key_v4=$(exec_on cuke-master conjur user rotate_api_key)
-ssl_cert_v4=$(exec_on cuke-master cat /opt/conjur/etc/ssl/ca.pem)
+if oss_only; then
```
Comment (izgeri) [inline _setup.sh]: same comment as above

File: conjurapi/authn/auth_token_test.go (hunk 2 above)
Comment (doodlesbykumbi) [reply conjurapi/authn/auth_token_test.go]: i thought of it like so:
+ an ad-hoc call to the api is overkill and forces this part of the code to need a v4 appliance to test
+ the token format is well defined so there's no loss in generality by using mocks
+ i think it's worth having these tests be well scoped. in this case to parsing a token and nothing more

**Pull request merged**